src/main/java/org/stellar/sdk/Auth.java

```diff
@@ -0,0 +1,303 @@
+package org.stellar.sdk;
+
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import org.stellar.sdk.scval.Scv;
+import org.stellar.sdk.xdr.EnvelopeType;
+import org.stellar.sdk.xdr.Hash;
+import org.stellar.sdk.xdr.HashIDPreimage;
+import org.stellar.sdk.xdr.Int64;
+import org.stellar.sdk.xdr.SCVal;
+import org.stellar.sdk.xdr.SorobanAddressCredentials;
+import org.stellar.sdk.xdr.SorobanAuthorizationEntry;
+import org.stellar.sdk.xdr.SorobanAuthorizedInvocation;
+import org.stellar.sdk.xdr.SorobanCredentials;
+import org.stellar.sdk.xdr.SorobanCredentialsType;
+import org.stellar.sdk.xdr.Uint32;
+import org.stellar.sdk.xdr.XdrUnsignedInteger;
+
+/** This class contains helper methods to sign {@link SorobanAuthorizationEntry}. */
+public class Auth {
+  /**
+   * Actually authorizes an existing authorization entry using the given the credentials and
```

Comment:
for consideration in a later pr, could remove this duplicated docs by using `@see`  instead:
```
/**
 * @see Auth#authorizeEntry(String, KeyPair, Long, Network) authorizeEntry
 */
```